### ISSUE TITLE
Cannot pull or clone from Enterprise GitHub using GitHub Desktop authentication fails

### ISSUE THREAD

**Issue by keaedwar** [desktop/desktop#10927]
### Describe the bug

I'm unable to clone or pull updates from an Enterprise GitHub repository with GitHub desktop. I have no problem using the git command line bash interface or the git GUI interface (the two tools installed as a part of the git installation). I started noticing problems on Monday, October 26, 2020, when I could no longer pull or clone repositories with GitHub Desktop from our Enterprise GitHub Repository. The Enterprise admins say they haven't made any changes. At one point they said that some new certificates may have been installed, but they don't know what the problem is. I've tried to log out and log back in to try to refresh, but that doesn't work.

I saw this in the log:

    2020-10-27T14:21:04.301Z - info: [ui] [AppStore.withAuthenticatingUser] account found for repository: AdminTools - keaedwar (has token)
    2020-10-27T14:21:05.101Z - error: [ui] `git -c credential.helper= fetch --progress --prune origin` exited with an unexpected code: 128.
    stderr:
    remote: Anonymous access denied

and I thought this might be the problem, but then I looked back in previous logs and they have similar errors and I was able to push, pull, clone without a problem previous to October 26th without noticing any problems or getting any authentication errors showing on GitHub Desktop.

I'm not sure what to do next to try to get the authentication working again.

### Version & OS

GitHub Desktop version 2.5.6
Windows 10 Enterprise Version 1809

### Steps to reproduce the behavior

1. Go to File>Clone repository...
2. Select "GitHub Enterprise Server"
3. Scroll through the list to find a repository and select one
4. press the Clone button

A dialog pops up saying Authentication failed, I enter my Username and Password and I get the dialog box again.

### Expected behavior

Usually, it just worked. I was never prompted to login again on pulls or clones, since I was already logged in.

### Actual behavior

It seems like I can see the "private" repositories on the enterprise GitHub server, but I can't pull or clone them (so it seems like at least some of the authentication is working to allow me to see the repositories, but it must be going through some other authentication mechanism to actually pull or clone the repositories.

### Screenshots

Adding screenshots of login window showing logged in and adding a screenshot of the Authentication failure message.

### Logs

I'm adding a zip file with all the logs.

### Additional context

[GitHubDesktoplogs.zip](https://github.com/desktop/desktop/files/5446123/GitHubDesktoplogs.zip)
![Capture00](https://user-images.githubusercontent.com/61704744/97319419-3ec67100-1843-11eb-8b52-1de44d53e488.PNG)
![Capture01](https://user-images.githubusercontent.com/61704744/97319433-41c16180-1843-11eb-8fc7-7e46fd3f3a55.PNG)

**Comment by keaedwar** [comment]
One other thing is the Application Authorized OAuth Apps has GitHub Desktop and GitHub Extension for Visual Studio.

![Capture02](https://user-images.githubusercontent.com/61704744/97324578-5ce2a000-1848-11eb-94f1-c5184246cddf.PNG)

**Comment by AndreiMaga** [comment]
I used to get this a lot when I worked in a corp with a proxy, which you probably use.
```remote: Anonymous access denied``` Is exactly that, the server refusing your request because it cannot authenticate the request.
Try to use SSH links instead of HTTPS ones, that used to do it for me, might be different for you but it does not hurt to try.

**Comment by keaedwar** [comment]
AFAIK, we don't use a proxy, but we have Group Policies, VPNs and Firewalls. The odd part is that I can run a GIT CLI clone command on the enterprise https://.../repo.git repo just fine. GitHub desktop can list the enterprise repos, but If I want it to clone or pull changes from the repo in GitHub Desktop, I have to change the url in the local repo using the "Repository/Repository Settings..." to change the url from an https url to a SSH git@github.iu.edu:REPO for GitHub Desktop to work. Only after that, will GitHub Desktop work.  I don't think I can setup GitHub Desktop to always use the SSH format. I have a work around, but it's a pain.  I wish I knew what was going on.

**Comment by AndreiMaga** [comment]
As @billygriffin pointed out in https://github.com/desktop/desktop/issues/2789#issuecomment-603508392 
try to run ```git config --global --unset http.proxy ``` and it could solve your problem.

**Comment by keaedwar** [comment]
Thanks for your suggestions.

I tried unsetting http.proxy, but it didn't seem to change GitHub Desktop (I did restart GitHub Desktop), but I started wondering about other configurations. So I took a screenshot of my global and local config.  I'm wondering if there's something in the cert or cred settings that doesn't jibe.  Especially the credential.https setting.

Here's the Global Config for Git. I ran a list before I ran the unset command. I didn't think I had a proxy, but I ran it anyway just to make sure.

![2020-11-11 08_36_37-Window-Git-Global-Config](https://user-images.githubusercontent.com/61704744/98819968-892b1e80-23fb-11eb-92dd-e8d4a4aceec6.png)

That made me think of the user settings, so I listed those out.  The cred and cert settings make me wonder if there's something that should be changed (credential.helper, credential.https, http.sslcainfo).

![2020-11-11 08_45_26-Window-Git-Config](https://user-images.githubusercontent.com/61704744/98820433-29814300-23fc-11eb-8540-571e575d6686.png)

**Comment by steveward** [comment]
@keaedwar are you still encountering this error in GitHub Desktop? If so, could you try running this command from the command line within your repository?
```
"C:\Users\<Your_Windows_User>\AppData\Local\GitHubDesktop\app-2.6.0\resources\app\git\cmd\git.exe" -c credential.helper= -c protocol.version=2 fetch --progress --prune origin
```

Be sure to change the Windows user, and let me know if that throws the same error. I'd also recommend verifying that GitHub Desktop has been approved in your organization (see [this article](https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-organizations-and-teams/approving-oauth-apps-for-your-organization). It looks like that is the case, but I just wanted to verify.

**Comment by kae36** [comment]
Hi @steveward thanks for helping.  I am still getting this error in GitHub Desktop.  I ran the command line and this is what I got:

`C:\Users\keaedwar\source\alan\Admintools>C:\Users\keaedwar\AppData\Local\GitHubDesktop\app-2.6.0\resources\app\git\cmd\git.exe -c credential.helper= -c protocol.version=2 fetch --progress --prune origin
error: failed to execute prompt script (exit code 255)
fatal: could not read Username for 'https://github.iu.edu': No such file or directory`

I read the article you mentioned, but my organization settings look different and I wasn't able to go past step 4.  I don't see the **Third-party access** menu item in the left panel. Below is what I see in my Organization profile.  

![2020-12-01 08_45_04-Organization profile — Mozilla Firefox](https://user-images.githubusercontent.com/1879639/100748926-fce49980-33b1-11eb-8573-06f0b1073dd4.png)

The OAuth Apps and GitHub Apps under Developer settings look like those are for requesting OAuth for applications that we have developed using the GitHub API. We have nothing listed there.  

The odd thing is that we have five users in our organization and three of the users have no problems accessing the project repositories using GitHub Desktop, but two users have problems. The only way that I've been able to access the repository is by using the Git CLI commands to clone the repository using the CLI "git clone https://github.iu.edu/Organization/projectname.git", I can then go into GitHub Desktop and change the Repository settings for Remote to "git@github.iu.edu:Organization/projectname.git" and I can use GitHub Desktop after I change the repository settings.

Thanks for your help.
